proj/src/main/webapp/chat.js

```diff
@@ -812,5 +895,10 @@ function addTag(tag, uid) {
   document.querySelector('.tag-container').insertBefore(tagContainer, tagInput)
 }
 
+
+
+
+exports.createOrJoinChat = createOrJoinChat;
```

Comment:
Only reason there are so many empty lines is to reduce future merge conflicts:
TODO: (low priority) reduce number of empty lines